data-prepper-plugins/saas-source-plugins/saas-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/saas_crawler/model/ItemInfo.java

```diff
@@ -0,0 +1,49 @@
+package org.opensearch.dataprepper.plugins.source.saas_crawler.model;
+
+
+import lombok.Getter;
+import lombok.NonNull;
+
+import java.util.Map;
+
+
+@Getter
+public abstract class ItemInfo {
```

Comment:
This should be a Java `interface`. It is more flexible and allows for adapting rather than having to convert.

```
public interface ItemInfo {
  String getItemId();
  Map<String, Object> getMetadata();
  ...
}
```

Reply:
Converted